### PR TITLE
Feature/2085/absolute path check

### PR DESCRIPTION
**Pull request by garyluu**
For #2085 

Workaround that makes sure absolute paths added resemble absolute paths.  Log warn message when workaround was used...this should be fixed correctly at some point.

Not sure if relevant but there are files being added that have an empty path and/or absolute path before this PR

Push cancelled because I don't care about it.

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java [hunk 1]
```diff
@@ -150,7 +155,11 @@ public String getAbsolutePath() {
     }
 
     public void setAbsolutePath(String absolutePath) {
-        this.absolutePath = absolutePath;
+        // TODO: Figure out the actual absolute path before this workaround
+        this.absolutePath = ZipSourceFileHelper.addLeadingSlashIfNecessary((absolutePath));
+        if (this.absolutePath.equals(absolutePath)) {
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java]: Shouldn't this be not equals?

Comment (garyluu) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java]: doh!

File: dockstore-webservice/src/main/resources/migrations.1.6.0.xml [hunk 2]
```diff
@@ -343,4 +343,9 @@
             UPDATE sourcefile SET absolutepath = CONCAT('/', absolutepath) WHERE absolutepath NOT LIKE '/%'
         </sql>
     </changeSet>
+    <changeSet id="checkAbsolutePath" author="gluu">
+        <sql dbms="postgresql">
+            ALTER TABLE sourcefile ADD CONSTRAINT absolutepath_check CHECK (absolutepath LIKE '/%' OR absolutepath='')
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/resources/migrations.1.6.0.xml]: Has this been tested with the most recent censored dump?

Comment (garyluu) [reply dockstore-webservice/src/main/resources/migrations.1.6.0.xml]: just tested again with:

```
psql webservice_test < ~/Downloads/ds-webservice*sql -U dockstore
java -jar ~/dockstore/dockstore-webservice/target/dockstore-webservice*SNAPSHOT.jar db migrate -i 1.5.0,1.6.0 ~/.dockstore/dockstore.yml
```

and ran webservice with validate, it seems to not complain.

**Pull request merged**